Source/santactl/Commands/sync/SNTCommandSyncRuleDownload.m

```diff
@@ -30,30 +30,16 @@ - (NSURL *)stageURL {
 }
 
 - (BOOL)sync {
-  self.syncState.downloadedRules = [NSMutableArray array];
-  return [self ruleDownloadWithCursor:nil];
-}
-
-- (BOOL)ruleDownloadWithCursor:(NSString *)cursor {
-  NSDictionary *requestDict = (cursor ? @{kCursor : cursor} : @{});
-
-  NSDictionary *resp = [self performRequest:[self requestWithDictionary:requestDict]];
-  if (!resp) return NO;
-
-  for (NSDictionary *rule in resp[kRules]) {
-    SNTRule *r = [self ruleFromDictionary:rule];
-    if (r) [self.syncState.downloadedRules addObject:r];
-  }
-
-  if (resp[kCursor]) {
-    return [self ruleDownloadWithCursor:resp[kCursor]];
-  }
-
-  if (!self.syncState.downloadedRules.count) return YES;
+  // Grab the new rules from server
+  NSArray<SNTRule *> *newRules = [self downloadNewRulesFromServer];
+  if (!newRules) return NO;         // encountered a problem with the download
+  if (!newRules.count) return YES;  // successfully downloaded rules, but nothing of interest
```

Comment:
Maybe something like:
successfully completed the request, but there are no new rules